nodejs/week2/app.js

```diff
@@ -0,0 +1,79 @@
+const { error } = require("console");
+const express = require("express");
+const app = express();
+const port = process.env.PORT || 3000;
+// Support parsing JSON requests
+app.use(express.json());
+// Importig our doc json
+const documents = require("./documents.json");
+// If q is not provided, the endpoint should return all documents.
+// If q is provided, the endpoint should return the documents with some field that matches the value of q.
+app.get("/", (req, res) => {
+  res.send("This is a search engine");
+});
+app.get("/search", (req, res) => {
+  try {
+    const userQuery = req.query.q.toLowerCase();
+    const filteredDocument = documents.filter((info) => {
+      for (const [key, value] of Object.entries(info)) {
+        const valueString = String(value).toLowerCase();
+        const keyString = String(key).toLowerCase();
+        if (keyString.includes(userQuery) || valueString.includes(userQuery)) {
+          return true;
+        }
+      }
+    });
+    res.json({ data: filteredDocument });
+  } catch (error) {
+    res.status(500).json({ error: error.message });
+  }
+});
+
+// GET /documents/:id
+app.get("/documents/:id", (req, res) => {
+  try {
+    const id = parseInt(req.params.id);
+    const filteredDocument = documents.filter((info) => {
+      for (const [key, value] of Object.entries(info)) {
+        if (value === id) {
+          return info;
+        }
+      }
+    });
```

Comment:
It works like a charm! You do not have to update this it's just a FYI.
There is a cool function called  [find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find)  that can cut this to two lines:
```
 const id = parseInt(req.params.id);
 const doc = documents.find(doc => doc.id == id);
```

Reply:
Thank you for your review. Array.methods are not my strongest side, agree. Need more practice in this :)